### DISCUSSION TITLE
Retrieving AuthUser from useAuthUser

### DISCUSSION THREAD

**Discussion by MattHeslington** [gladly-team/next-firebase-auth#263] [Q&A] [answered]
Hello Gladly,

Thanks for the best Next auth solution!

I've got what I think is a really simple problem. Basically, what is the 'recommended' way to check if there is a logged-in user? 

Specifically, I have an 'Upvote' component. It must check whether there is a signed in user before running a API fetch in a useEffect. In the example below, AuthUser.email always returns null when I am logged in, I can't understand why. The component is wrapped in 'withAuthUser'. I've tried wrapping parent components and the parent page with withAuthUser as well but that doesn't help.

`useEffect(async () => {
    ...
    if (AuthUser.email) {
        try {
            ...
        }
    }
},[])`

Many thanks for any advice.
Cheers, Matt

**Comment by shreyas-jadhav** [comment]
Maybe the function could he running before initialisation. Have to tried to wait until `clientInitialized` becomes `true`? 🤔

**Comment by kmjennison** [comment, marked as answer]
Hi Matt, thanks!

To check if there's a logged-in user, I recommend checking if `AuthUser.id` is set. Also, your `useEffect` hook is missing `AuthUser` as a dependency, so like @shreyas-jadhav mentioned, it's probably running before the Firebase client has initialized.

If it's still not working, please try running the example app in this repository with your own Firebase settings to see if you can get that working.

**Reply by MattHeslington** [reply]
Cheers, Kevin, Shreyas. There was actually something wrong in my useEffect which caused it to break when I added AuthUser as a dependancy. All working well now.
